pkg/curatedpackages/packagecontrollerclient.go

```diff
@@ -335,3 +348,71 @@ func WithValuesFileWriter(writer filewriter.FileWriter) func(client *PackageCont
 		config.valuesFileWriter = writer
 	}
 }
+
+func formatYamlLine(space, key, value string) string {
+	if value == "" {
+		return ""
+	}
+	return space + key + ": " + value + "\n"
+}
+
+func formatImageResource(resource *v1alpha1.ImageResource, name string) (result string) {
+	if resource.CPU != "" || resource.Memory != "" {
+		result = "    " + name + ":\n"
+		result += formatYamlLine("      ", "cpu", resource.CPU)
+		result += formatYamlLine("      ", "memory", resource.Memory)
+	}
+	return result
+}
+
+func formatCronJob(cronJob *v1alpha1.PackageControllerCronJob) (result string) {
+	if cronJob != nil {
+		result += "cronjob:\n"
+		result += formatYamlLine("  ", "digest", cronJob.Digest)
+		result += formatYamlLine("  ", "repository", cronJob.Repository)
+		result += formatYamlLine("  ", "suspend", strconv.FormatBool(cronJob.Disable))
+		result += formatYamlLine("  ", "tag", cronJob.Tag)
+	}
+	return result
+}
+
+func formatResources(resources *v1alpha1.PackageControllerResources) (result string) {
+	if resources.Limits.CPU != "" || resources.Limits.Memory != "" ||
+		resources.Requests.CPU != "" || resources.Requests.Memory != "" {
+		result += "  resources:\n"
+		result += formatImageResource(&resources.Limits, "limits")
+		result += formatImageResource(&resources.Requests, "requests")
+	}
+	return result
+}
+
+// GetPackageControllerConfiguration returns the default kubernetes version for a Cluster.
+func GetPackageControllerConfiguration(clusterSpec *v1alpha1.ClusterSpec) (result string, err error) {
```

Comment:
Make this a method and then u won't need the clusterSpec parameter.

Reply:
I think in the long road, PCC is not the place for this. Refactoring a solution between PackageInstaller and PCC is out of scope for this effort and I'd sooner make as few changes here to minimize the chances of breaking something right now.

Reply:
I don't understand why a refactor is needed to make this change?

Reply:
This could be added to the PCC easy enough, but I don't think that is the right place for it. There should be a new class created that combines the some of the logic of PCC and PackageInstaller. I don't want to start ripping all this apart right now.

Reply:
nvm, I moved it over there